### PR TITLE
#45 iso.Key.nearest_note - upper bound issue - fix to amended test

### PR DESCRIPTION
**Pull request by piotereks**

### CODE REVIEW

File: isobar/key.py [hunk 1]
```diff
@@ -65,11 +65,18 @@ def nearest_note(self, note):
...
-            for semi in self.semitones:
-                dist = abs(semi - pitch)
+            chk_semitones = self.semitones
+            calc_octave = octave
+            for semi in chk_semitones:
+                dist = min(abs(semi - pitch),abs(abs(semi-pitch)-12))
                 if nearest_dist is None or dist < nearest_dist:
                     nearest_semi = semi
                     nearest_dist = dist
+                    if dist == abs(abs(semi-pitch)-12):
```
Comment (ideoforms) [inline isobar/key.py:75]: Hardcoded `12` here should be `self.scale.octave_size`, then good to merge!

**Review by ideoforms — CHANGES_REQUESTED** [review]
Thanks for this patch and sorry for the slow reply! I've only had limited bandwidth to deal with PRs over the past few months. One change then good to go.

**Review by ideoforms — APPROVED** [review]

**Comment by ideoforms** [comment]
@piotereks Unit test now seems to be failing 🤔

**Comment by piotereks** [comment]
Fixed with other commit. Nearest note was faulty because of 2 additional
things:
- non deterministic selection of note in case lowe and upper note have same
distance
- nearest note was disregarding other than 0 tonic for scale

Pozdrawiam
Piotr

śr., 3 sty 2024, 22:01 użytkownik Daniel Jones ***@***.***>
napisał:

**Pull request merged**